app/src/main/cpp/skyline/gpu/graphics_pipeline_assembler.cpp

```diff
@@ -252,4 +255,8 @@ namespace skyline::gpu {
             SerialisePipelineCache(gpu, pipelineCacheDir, rawData);
         });
     }
+
+    void GraphicsPipelineAssembler::RegisterCompilationCallback(std::function<void()> callback) {
+        compilationCallback = std::move(callback);
+    }
```

Comment:
Make unregistering a seperate function, and throw exception here if a callback is already registered

Reply:
IMO we should add an override boolean parameter defaulted to false instead of throwing an exception

Reply:
Nah, that's confusing

Reply:
Ok